src/problem.cpp

```diff
@@ -142,6 +144,43 @@ const TensorDescriptor& Problem::GetTensorDescriptorChecked(miopenTensorArgument
     return found->second;
 }
 
+Problem Problem::Transpose() const
```

Comment:
```suggestion
Problem Problem::MakeTransposed() const
```
or maybe even `ConstructTransposed`. `Transpose` suggests that this method modifies existing object, which is not so.